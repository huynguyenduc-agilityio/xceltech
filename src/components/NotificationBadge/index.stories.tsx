import type { Meta, StoryObj } from '@storybook/react';

// Icons
import { Bell } from 'lucide-react';

// Components
import NotificationBadge from '.';

const meta = {
  title: 'Components/NotificationBadge',
  component: NotificationBadge,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <NotificationBadge {...args} />
    </div>
  ),
} satisfies Meta<typeof NotificationBadge>;

export default meta;

type Story = StoryObj<typeof NotificationBadge>;

export const Default: Story = {
  args: { count: 13 },
};

export const WithIcon: Story = {
  args: { icon: <Bell className="text-white w-6 h-6" />, count: 13 },
};
