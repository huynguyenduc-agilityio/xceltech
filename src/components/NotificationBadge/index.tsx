import { ReactNode } from 'react';
import { cn } from '@/utils';

export interface NotificationBadgeProps {
  count: number;
  icon?: ReactNode;
  className?: string;
}

const NotificationBadge = ({
  count,
  icon,
  className,
}: NotificationBadgeProps) => {
  if (count <= 0) return null;

  return icon ? (
    <div
      className={cn(
        'relative flex items-center justify-center rounded-full w-[53px] h-[53px] bg-primary',
        className,
      )}
    >
      {icon}
      <div className="absolute -top-0 -right-2">
        <div
          className={cn(
            'flex h-5 w-5 items-center justify-center rounded-full border border-white bg-red-primary text-white text-xs font-bold shadow-md',
            !icon && className,
          )}
        >
          {count}
        </div>
      </div>
    </div>
  ) : (
    <div
      className={cn(
        'flex h-5 w-5 items-center justify-center rounded-full border border-white bg-red-primary text-white text-xs font-bold shadow-md',
        !icon && className,
      )}
    >
      {count}
    </div>
  );
};

export default NotificationBadge;
