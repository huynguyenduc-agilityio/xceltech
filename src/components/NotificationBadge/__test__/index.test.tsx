import { render } from '@testing-library/react';

// Components
import NotificationBadge from '..';
import { Bell } from 'lucide-react';

describe('NotificationBadge component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <NotificationBadge
        icon={<Bell className="text-white w-6 h-6" />}
        count={13}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
