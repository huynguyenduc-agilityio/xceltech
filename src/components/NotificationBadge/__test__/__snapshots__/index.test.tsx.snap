// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBadge component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative flex items-center justify-center rounded-full w-[53px] h-[53px] bg-primary"
  >
    <svg
      class="lucide lucide-bell text-white w-6 h-6"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.268 21a2 2 0 0 0 3.464 0"
      />
      <path
        d="M3.262 15.326A1 1 0 0 0 4 17h16a1 1 0 0 0 .74-1.673C19.41 13.956 18 12.499 18 8A6 6 0 0 0 6 8c0 4.499-1.411 5.956-2.738 7.326"
      />
    </svg>
    <div
      class="absolute -top-0 -right-2"
    >
      <div
        class="flex h-5 w-5 items-center justify-center rounded-full border border-white bg-red-primary text-white text-xs font-bold shadow-md"
      >
        13
      </div>
    </div>
  </div>
</DocumentFragment>
`;
